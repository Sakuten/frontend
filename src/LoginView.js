import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';

@inject('event')
@observer
export class LoginView extends Component {
  render() {
    return (
      <div>
        <input type="text" name="username" value={this.props.credential.username} onChange={e => { this.props.credential.setUsername(e.target.value) }} />
        <input type="text" name="password" value={this.props.credential.password} onChange={e => { this.props.credential.setPassword(e.target.value) }} />
        <button onClick={this.props.event.credential.login}>Login</button>
      </div>
    );
  }
}
