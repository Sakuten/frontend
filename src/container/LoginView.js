import React from 'react'
import { inject, observer } from 'mobx-react'
import ReCAPTCHA from 'react-google-recaptcha'
import QRReader from '../component/QRReader'
import styled from 'styled-components'

const Container = styled.div`
  width: 90vw;
  height: 90vw;

  display: flex;
  justify-content: center;
  align-items: center;
`

const LoginView = ({credential, event}) => {
  const {
    onQRError,
    onQRScan,
    onChangeRecaptchaResponse
  } = event.credential

  let captcha

  return (
    <div data-test='loginview'>
      <Container>
        {
          credential.secretId
            ? <ReCAPTCHA
              size='normal'
              sitekey={process.env.REACT_APP_RECAPTCHA_KEY}
              onChange={onChangeRecaptchaResponse}
              onExpired={() => captcha.reset()}
              ref={(el) => { captcha = el }}
            />
            : <QRReader
              onError={onQRError}
              onScan={onQRScan}
            />
        }
      </Container>
    </div>
  )
}

export default inject('event')(observer(LoginView))
