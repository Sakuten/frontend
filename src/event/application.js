import {applyLottery, cancelLottery} from '../api/operation'

export class ApplicationObject {
  constructor (store) {
    this.store = store
    this.onUpdate()
  }

  onUpdate = async () => {
    await Promise.all([
      this.store.application.fetchClassroomList(),
      this.store.application.fetchLotteryList()
    ])
  }

  onChangeClassroom = (classroom) => {
    this.store.application.setClassroom(classroom)
  }

  onChangeLottery = (lottery) => {
    this.store.application.setLottery(lottery)
  }

  onAddGroupMember = (secretId) => {
    this.store.application.addGroupMember(secretId)
  }

  onRemoveGroupMember = (idx) => {
    this.store.application.removeGroupMemberWithIdx(idx)
  }

  onApply = async () => {
    await applyLottery(this.store.application.lottery, this.store.application.groupMemberList, this.store.credential.token)
    await this.store.fetchStatus()
  }

  onCancel = async (id) => {
    await cancelLottery(id, this.store.credential.token)
    await this.store.fetchStatus()
  }
}
