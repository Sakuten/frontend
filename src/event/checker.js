import {getPublicId, checkSecretIdStatus} from '../api/operation'
import {extractId} from '../util/extractId'

export class CheckerObject {
  constructor (store) {
    this.store = store
    this.lastScan = ''
  }

  onChangeClassroom = (classroom) => {
    this.store.checker.setClassroom(classroom)
  }

  onQRScan = async (scanUri) => {
    if (scanUri) {
      if (this.lastScan === scanUri) {
        return
      }
      this.lastScan = scanUri
      const secretId = extractId(scanUri)
      if (!secretId) {
        this.store.error.addError(101, 'Invalid QR Code')
        return
      }

      this.store.checker.setLastStatus('取得中')
      this.store.checker.setPublicId('')

      const status = await this.store.error.ignoring([19, 6], (callback) => {
        return checkSecretIdStatus(this.store.checker.classroom, secretId, this.store.credential.token)
          .then(resp => {
            return resp.data['status']
          })
          .catch(e => {
            const codeStatuses = {
              19: '応募していません',
              6: '時間外です'
            }
            return codeStatuses[e.response.data.code] || JSON.stringify(e.response.data)
          })
          .then(callback)
      })

      const resp = await getPublicId(secretId, this.store.credential.token)

      this.store.checker.setLastStatus(status)
      this.store.checker.setPublicId(resp.data['public_id'])
    }
  }
}
