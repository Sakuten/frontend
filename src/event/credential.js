import {fetchApi} from '../api'

export class CredentialObject {
  constructor(store) {
    this.store = store;
  }

  login = () => {
    fetchApi('auth/', {
      method: 'post',
      headers: {
        'Content-Type': 'application/json'
      },
      data: {
        password: this.store.credential.password,
        username: this.store.credential.username
      }
    }).then(async (response) => {
      const json = response.data
      if ('token' in json) {
        this.store.credential.setToken(json.token)
        this.store.credential.clearPassword()
        this.fetchStatus()
      } else { throw Error('Invalid response returned') }
    })
  }

  fetchStatus = () => {
    fetchApi(`api/status`, {
      method: 'get',
      headers: {
        'Authorization': 'Bearer ' + this.store.credential.token
      }
    })
      .then(response => {
        this.store.credential.setStatus(response.data.status)
      })
      .catch(error => {
        if (error.response && error.response.data.message === 'Unauthorized') {
          this.logout()
        } else {
          throw error
        }
      })
  }

  logout = () => {
    this.store.credential.setToken('')
  }

  onChangeUsername = (username) => {
    username = username.trim()
    this.store.credential.setUsername(username)
  }

  onChangePassword = (password) => {
    password = password.trim()
    this.store.credential.setPassword(password)
  }
}
