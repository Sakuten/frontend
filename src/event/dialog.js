export class DialogObject {
  constructor (store) {
    this.store = store
  }

  onClose = () => {
    this.store.dialog.close()
  }

  onOpen = (title, content, buttonText = 'OK', enableButton = true) => {
    this.store.dialog.setTitle(title)
    this.store.dialog.setContent(content)
    this.store.dialog.setButtonText(buttonText)
    this.store.dialog.setButtonEnabled(enableButton)
    this.store.dialog.open()
  }
}
