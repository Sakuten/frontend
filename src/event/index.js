import {CredentialObject} from './credential'
import {ApplicationObject} from './application'
import {CheckerObject} from './checker'
import {ErrorObject} from './error'
import {DialogObject} from './dialog'

import {applyLottery, cancelLottery} from '../api/operation'

export class Event {
  constructor (store) {
    this.store = store
    this.error = new ErrorObject(store)
    this.credential = new CredentialObject(store)
    this.application = new ApplicationObject(store)
    this.dialog = new DialogObject(store)
    this.checker = new CheckerObject(store)

    // This can cause network error, to be caught in ErrorObject
    // So fetchStatus() have to be placed in here
    if (this.store.credential.isLoggedIn) { this.store.fetchStatus() }
  }

  onApplyLottery = async () => {
    this.dialog.onOpen('応募しています', 'しばらくお待ちください', 'お待ちください', false)
    try {
      await applyLottery(this.store.application.lottery, this.store.application.groupMemberList.map(pair => pair[0]), this.store.credential.token)
    } catch (err) {
      this.dialog.onClose()
      throw err
    }
    this.dialog.onOpen('応募しました', '発表をお待ちください', 'OK')
    await this.store.fetchStatus()
    if (this.store.credential.isUsedByStaff) {
      this.store.credential.logout()
    }
  }

  onCancelApplication = async (id) => {
    this.dialog.onOpen('キャンセルしています', 'しばらくお待ちください', 'お待ちください', false)
    try {
      await cancelLottery(id, this.store.credential.token)
    } catch (err) {
      this.dialog.onClose()
      throw err
    }
    this.dialog.onOpen('キャンセルしました', '応募は取り消されました', 'OK')
    await this.store.fetchStatus()
    if (this.store.credential.isUsedByStaff) {
      this.store.credential.logout()
    }
  }
}
