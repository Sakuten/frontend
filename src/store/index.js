import {CredentialObject} from './credential'
import {ApplicationObject} from './application'
import {CheckerObject} from './checker'
import {DialogObject} from './dialog'
import { RouterStore } from 'mobx-react-router'
import {ErrorObject} from './error'

export class Store {
  constructor () {
    this.credential = new CredentialObject()
    this.application = new ApplicationObject()
    this.checker = new CheckerObject()
    this.dialog = new DialogObject()
    this.router = new RouterStore()
    this.error = new ErrorObject()
  }

  // Called in Event's constructor
  fetchStatus = async () => {
    await this.credential.fetchStatus()
    if (!this.credential.isLoggedInAsChecker) {
      await this.application.fetchApplicationList(this.credential.token)
    }
  }
}
