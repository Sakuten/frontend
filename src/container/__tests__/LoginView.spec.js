import React from 'react'
import { shallow, mount } from 'enzyme'
import deepAssign from 'deep-assign'
import { Provider } from 'mobx-react'
import LoginView from '../LoginView'
import {Store} from '../../store'
import {Event} from '../../event'

const setup = (propOverrides, storeOverrides, eventOverrides, isShallow = true) => {
  const store = deepAssign(new Store(), storeOverrides)
  const event = deepAssign(new Event(store), eventOverrides)
  const props = Object.assign({
    credential: store.credential
  }, propOverrides)

  const wrapper = (isShallow ? shallow : mount)(<Provider event={event}><LoginView {...props} /></Provider>)

  return {
    props,
    wrapper,
    store,
    event,
    usernameInput: wrapper.find('[data-test="loginview-username"]'),
    passwordInput: wrapper.find('[data-test="loginview-password"]'),
    loginButton: wrapper.find('[data-test="loginview-login"]')
  }
}

describe('containers', () => {
  describe('LoginView', () => {
    it('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders username stored in store', () => {
      const { usernameInput } = setup({}, {credential: {username: 'username'}}, {}, false)
      expect(usernameInput.props().value).toBe('username')
    })

    it('renders password stored in store', () => {
      const { passwordInput } = setup({}, {credential: {password: 'password'}}, {}, false)
      expect(passwordInput.props().value).toBe('password')
    })
  })
})
