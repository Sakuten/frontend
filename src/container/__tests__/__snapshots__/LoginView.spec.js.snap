// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers LoginView render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    event={
      Event {
        "application": ApplicationObject {
          "onAddGroupMember": [Function],
          "onChangeClassroom": [Function],
          "onChangeLottery": [Function],
          "onRemoveGroupMember": [Function],
          "onUpdate": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "checker": CheckerObject {},
            "credential": CredentialObject {
              "kind": "",
              "recaptchaResponse": "",
              "secretId": "",
              "status": Object {},
              "token": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "secretId" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "kind" => "",
                  "status" => Map {},
                  "isUsedByStaff" => false,
                  "isLoggedIn" => false,
                  "isAbleToAuthenicate" => false,
                  "isLoggedInAsChecker" => false,
                },
              },
            },
            "dialog": DialogObject {},
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "checker": CheckerObject {
          "lastScan": "",
          "onChangeClassroom": [Function],
          "onQRScan": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "checker": CheckerObject {},
            "credential": CredentialObject {
              "kind": "",
              "recaptchaResponse": "",
              "secretId": "",
              "status": Object {},
              "token": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "secretId" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "kind" => "",
                  "status" => Map {},
                  "isUsedByStaff" => false,
                  "isLoggedIn" => false,
                  "isAbleToAuthenicate" => false,
                  "isLoggedInAsChecker" => false,
                },
              },
            },
            "dialog": DialogObject {},
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "credential": CredentialObject {
          "onChangeRecaptchaResponse": [Function],
          "onLogin": [Function],
          "onLogout": [Function],
          "onQRError": [Function],
          "onQRScan": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "checker": CheckerObject {},
            "credential": CredentialObject {
              "kind": "",
              "recaptchaResponse": "",
              "secretId": "",
              "status": Object {},
              "token": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "secretId" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "kind" => "",
                  "status" => Map {},
                  "isUsedByStaff" => false,
                  "isLoggedIn" => false,
                  "isAbleToAuthenicate" => false,
                  "isLoggedInAsChecker" => false,
                },
              },
            },
            "dialog": DialogObject {},
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "dialog": DialogObject {
          "onClose": [Function],
          "onOpen": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "checker": CheckerObject {},
            "credential": CredentialObject {
              "kind": "",
              "recaptchaResponse": "",
              "secretId": "",
              "status": Object {},
              "token": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "secretId" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "kind" => "",
                  "status" => Map {},
                  "isUsedByStaff" => false,
                  "isLoggedIn" => false,
                  "isAbleToAuthenicate" => false,
                  "isLoggedInAsChecker" => false,
                },
              },
            },
            "dialog": DialogObject {},
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "error": ErrorObject {
          "onDelete": [Function],
          "onError": [Function],
          "store": Store {
            "application": ApplicationObject {},
            "checker": CheckerObject {},
            "credential": CredentialObject {
              "kind": "",
              "recaptchaResponse": "",
              "secretId": "",
              "status": Object {},
              "token": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "CredentialObject@3.keys",
                  "observers": Set {},
                },
                "name": "CredentialObject@3",
                "target": [Circular],
                "values": Map {
                  "secretId" => "",
                  "recaptchaResponse" => "",
                  "token" => "",
                  "kind" => "",
                  "status" => Map {},
                  "isUsedByStaff" => false,
                  "isLoggedIn" => false,
                  "isAbleToAuthenicate" => false,
                  "isLoggedInAsChecker" => false,
                },
              },
            },
            "dialog": DialogObject {},
            "error": ErrorObject {},
            "fetchStatus": [Function],
            "router": RouterStore {
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "history": null,
              "push": [Function],
              "replace": [Function],
            },
          },
        },
        "onApplyLottery": [Function],
        "onCancelApplication": [Function],
        "store": Store {
          "application": ApplicationObject {},
          "checker": CheckerObject {},
          "credential": CredentialObject {
            "kind": "",
            "recaptchaResponse": "",
            "secretId": "",
            "status": Object {},
            "token": "",
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "CredentialObject@3.keys",
                "observers": Set {},
              },
              "name": "CredentialObject@3",
              "target": [Circular],
              "values": Map {
                "secretId" => "",
                "recaptchaResponse" => "",
                "token" => "",
                "kind" => "",
                "status" => Map {},
                "isUsedByStaff" => false,
                "isLoggedIn" => false,
                "isAbleToAuthenicate" => false,
                "isLoggedInAsChecker" => false,
              },
            },
          },
          "dialog": DialogObject {},
          "error": ErrorObject {},
          "fetchStatus": [Function],
          "router": RouterStore {
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "history": null,
            "push": [Function],
            "replace": [Function],
          },
        },
      }
    }
  >
    <inject-LoginView-with-event
      credential={
        CredentialObject {
          "kind": "",
          "recaptchaResponse": "",
          "secretId": "",
          "status": Object {},
          "token": "",
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "CredentialObject@3.keys",
              "observers": Set {},
            },
            "name": "CredentialObject@3",
            "target": [Circular],
            "values": Map {
              "secretId" => "",
              "recaptchaResponse" => "",
              "token" => "",
              "kind" => "",
              "status" => Map {},
              "isUsedByStaff" => false,
              "isLoggedIn" => false,
              "isAbleToAuthenicate" => false,
              "isLoggedInAsChecker" => false,
            },
          },
        }
      }
    />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "credential": CredentialObject {
        "kind": "",
        "recaptchaResponse": "",
        "secretId": "",
        "status": Object {},
        "token": "",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "CredentialObject@3.keys",
            "observers": Set {},
          },
          "name": "CredentialObject@3",
          "target": [Circular],
          "values": Map {
            "secretId" => "",
            "recaptchaResponse" => "",
            "token" => "",
            "kind" => "",
            "status" => Map {},
            "isUsedByStaff" => false,
            "isLoggedIn" => false,
            "isAbleToAuthenicate" => false,
            "isLoggedInAsChecker" => false,
          },
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "credential": CredentialObject {
          "kind": "",
          "recaptchaResponse": "",
          "secretId": "",
          "status": Object {},
          "token": "",
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "CredentialObject@3.keys",
              "observers": Set {},
            },
            "name": "CredentialObject@3",
            "target": [Circular],
            "values": Map {
              "secretId" => "",
              "recaptchaResponse" => "",
              "token" => "",
              "kind" => "",
              "status" => Map {},
              "isUsedByStaff" => false,
              "isLoggedIn" => false,
              "isAbleToAuthenicate" => false,
              "isLoggedInAsChecker" => false,
            },
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
